include/openssl/posix_time.h

```diff
@@ -0,0 +1,48 @@
+/*
```

Comment:
I don't think we want new public APIs if we can avoid them, they are way too difficult remove or update according to the stability policy.  As a rule, we generally make things internal initially and expose them later if there is demand.

Reply:
Many apps have to carry around the equivalent of what @bob-beck is adding here.
It will also ease usage across the forks which is an added bonus.

Reply:
I disagree.

1. This is a notoriously difficult function to implement correctly.  Why do we think we'll get it right?  Are we better than the experts at this?
2. This version appears to have been implemented by a non expert given the computation epoch used.
3. I already know of inputs where this will produce incorrect results: it is already flawed.
4. This really is more difficult that it first appears.
5. If this is a public API it must work for all inputs.
6. This is not something cryptography experts should be attempting.